service/efsPort.py

```diff
@@ -0,0 +1,630 @@
+import inspect
+import os
+import platform
+import re
+import sys
+import traceback
+import json
+import eos.db
+
+from math import log
+from service.fit import Fit
+from service.market import Market
+from eos.enum import Enum
+from eos.saveddata.module import Hardpoint, Slot, Module
+from eos.saveddata.drone import Drone
+from eos.effectHandlerHelpers import HandledList
+from eos.db import gamedata_session, getItemsByCategory, getCategory, getAttributeInfo, getGroup
+from eos.gamedata import Category, Group, Item, Traits, Attribute, Effect, ItemEffect
+from logbook import Logger
+pyfalog = Logger(__name__)
+eos.db.saveddata_meta.create_all()
+
+
+class RigSize(Enum):
+    # Matches to item attribute "rigSize" on ship and rig items
+    SMALL = 1
+    MEDIUM = 2
+    LARGE = 3
+    CAPITAL = 4
+
+
+class EfsPort():
+    wepTestSet = {}
+
+    @staticmethod
+    def attrDirectMap(values, target, source):
+        for val in values:
+            target[val] = source.itemModifiedAttributes[val]
+
+    @staticmethod
+    def getT2MwdSpeed(fit, sFit):
+        fitID = fit.ID
+        propID = None
+        shipHasMedSlots = fit.ship.itemModifiedAttributes["medSlots"] > 0
+        shipPower = fit.ship.itemModifiedAttributes["powerOutput"]
+        # Monitors have a 99% reduction to prop mod power requirements
+        if fit.ship.name == "Monitor":
+            shipPower *= 100
```

Comment:
Looks like we're attempting to do some fitting calculations here for the Monitor...? Is this something that isn't implemented in the regular pyfa calcs?

Reply:
The shipPower is being compared to the base power of MWDs to work out if the ship could plausibly fit them.
We would have to actually equip each option for pyfa's calcs to apply which would be much, much slower.
If the bonus becomes widespread I'll certainly want to change the implementation.

Reply:
kk, I think I'm picking up what you're putting down. It's just another case of me not fully understanding how EFS is using this data. 😛  Thanks for the clarification